packages/functions/src/resample.ts

```diff
@@ -46,14 +47,40 @@ export function resample(_options: ResampleOptions = RESAMPLE_DEFAULTS): Transfo
 			}
 
 			for (const sampler of animation.listSamplers()) {
-				if (samplerTargetPaths.get(sampler) === 'weights') {
+				const targetPath = samplerTargetPaths.get(sampler)!;
+				if (targetPath === 'weights') {
 					didSkipMorphTargets = true;
 					continue;
 				}
-				if (sampler.getInterpolation() === 'STEP' || sampler.getInterpolation() === 'LINEAR') {
-					accessorsVisited.add(sampler.getInput()!);
-					accessorsVisited.add(sampler.getOutput()!);
-					optimize(sampler, samplerTargetPaths.get(sampler)!, options);
+
+				const samplerInterpolation = sampler.getInterpolation();
+
+				if (samplerInterpolation === 'STEP' || samplerInterpolation === 'LINEAR') {
+					const input = sampler.getInput()!;
+					const output = sampler.getOutput()!;
+
+					accessorsVisited.add(input);
+					accessorsVisited.add(output);
+
+					const times = input.getArray() as Float32Array;
+					const values = output.getArray() as Float32Array;
+					const elementSize = values.length / times.length;
+
+					const srcCount = times.length;
+					let dstCount: number;
+
+					if (samplerInterpolation === 'STEP') {
+						dstCount = resampleWASM(times, values, 'step');
+					} else if (targetPath === 'rotation') {
+						dstCount = resampleWASM(times, values, 'slerp');
+					} else {
+						dstCount = resampleWASM(times, values, 'lerp');
+					}
+
+					if (dstCount < srcCount) {
+						sampler.setInput(input.clone().setArray(times.slice(0, dstCount)));
+						sampler.setOutput(output.clone().setArray(values.slice(0, dstCount * elementSize)));
```

Comment:
TODO: The original arrays have already been overwritten at this point. Decide if we want to preemptively clone the arrays in JS before resampling, or to change the WASM API.

Reply:
Perf note here: `input.clone()` would clone the underlying array, to avoid that:
```javascript
const array = input.getArray();
input.setArray(null);
const cloned = input.clone();
input.setArray(array);
```

Also, it should be possible to detect if `input` and `output` need to be cloned, by checking existance of other usages from `listParents`.